### PR TITLE
Use either TERM or TERM_PROGRAM to detect wezterm

### PR DESCRIPTION
**Pull request by SuperSandro2000**

### CODE REVIEW

File: src/terminal.rs [hunk 1]
```diff
@@ -82,7 +82,8 @@ fn get_vte_version() -> Option<(u8, u8)> {
 
 /// Checks if the current terminal is WezTerm.
 fn is_wezterm() -> bool {
-    std::env::var("TERM_PROGRAM").map_or(false, |value| value == "WezTerm")
+    std::env::var("TERM").map_or(false, |value| value == "wezterm")
+    | std::env::var("TERM_PROGRAM").map_or(false, |value| value == "WezTerm")
```
Comment (lunaryorn) [inline src/terminal.rs:86]: Please swap the arms; I think we should check `TERM_PROGRAM` first, because it works out of the box in all local sessions.  We only really need to look at `TERM` over SSH sessions, and even then it only works if the user explicitly set it up.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,9 @@ To publish a new release run `scripts/release` from the project directory.
 
 ## [Unreleased]
 
+### Changed
+- Use either the default `TERM` enviroment variable or the none default `TERM_PROGRAM` for determining WezTerm terminal (see [GH-185]).
```
Comment (lunaryorn) [inline CHANGELOG.md:12]: Please rephrase this to remove the judgemental undertone; `TERM_PROGRAM` is not any less "default" then `TERM`.

Just say that mdcat now checks `TERM` in addition to `TERM_PROGRAM` to identify wezterm.

And please link to this merge request, e.g. GH-186, because that's the PR which actually implements it.

File: src/terminal.rs (hunk 1 above)
Comment (lunaryorn) [inline src/terminal.rs:86]: Please use `||`; `|` is a non-circuiting bitwise or.

**Review by lunaryorn — CHANGES_REQUESTED** [review]
In addition to my remarks directly in the code please also update the "Environment" section in `mdcat.1.adoc` so that it accurately describes how mdcat identifies terminals, and please take a look at the rustfmt issue.

File: CHANGELOG.md (hunk 2 above)
Comment (SuperSandro2000) [reply CHANGELOG.md:12]: According to the comment in wezterm it is not really a standard https://github.com/wez/wezterm/blob/ac5199c21663e39dcce887308a92b5e36b338866/config/src/lib.rs#L1709-L1710

Comment (lunaryorn) [reply CHANGELOG.md]: Well it speaks of an "emerging de facto standard" which is good enough for me 🙂 More so if it actually works by default 😜

And even if it wasn't we'd be using a nonstandard variable to detect a single nonstandard terminal emulator. I don't really see an issue with that 🙂

Comment (SuperSandro2000) [reply CHANGELOG.md]: Yeah, well whatever.

File: mdcat.1.adoc [hunk 3]
```diff
@@ -102,6 +102,7 @@ If run as `mdless` or if `--paginate` is given and the pager fails to start mdca
 == Environment
 
 TERM::
+    If this variable is `wezterm`, mdcat assumes that the terminal is WezTerm.
     If this variable is `xterm-kitty`, assume that the terminal is Kitty.
```
Comment (MuhammedZakir) [inline mdcat.1.adoc:106]:
```suggestion
    If this variable is `xterm-kitty`, mdcat assumes that the terminal is Kitty.
```

For consistency. :-)

**Review by lunaryorn — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,11 @@ To publish a new release run `scripts/release` from the project directory.
 
 ## [Unreleased]
 
+### Changed
+- WezTerm is now detected by `TERM` and `TERM_PROGRAM` enviroment variable (see [GH-186]).
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply CHANGELOG.md:12]:
```suggestion
- WezTerm is now detected by `TERM` and `TERM_PROGRAM` environment variables (see [GH-186]).
```

File: mdcat.1.adoc [hunk 5]
```diff
@@ -102,6 +102,7 @@ If run as `mdless` or if `--paginate` is given and the pager fails to start mdca
 == Environment
 
 TERM::
+    If this variable is `wezterm`, mdcat assumes that the terminal is WezTerm.
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply mdcat.1.adoc:105]: > If TERM_PROGRAM doesn't indicate wezterm (see below) and this variable is …"?

But TERM_PROGRAM can be anything as long as TERM is wezterm.

> So it's clear that you don't have to set both variables?

Both variables will be set as long as you are not in WSL or sshing into another machine.

[the remark this answers is not in the document]
Comment (lunaryorn) [reply mdcat.1.adoc:105]: In the current implementation mdcat will only look at `TERM` if `TERM_PROGRAM` doesn't already indicate wezterm.

And in my experience and according to its own documentation wezterm sets TERM to `xterm-256color` in order to avoid extra terminfo files, see e.g. the [FAQ](https://wezfurlong.org/wezterm/faq.html#how-to-troubleshoot-keys-that-dont-work-or-produce-weird-characters) (emphasis mime):

> Interactive Unix programs generally depend upon the TERM environment variable being set appropriately. **wezterm sets this to xterm-256color by default**, because wezterm aims to be compatible with with the settings defined by that terminfo entry. Setting TERM to something else can change the byte sequences that interactive applications expect to see for some keys, effectively disabling those keys.

So as far as I can see TERM cannot identify wezterm sufficiently in its default configuration which is why I prefer for mdcat to look at TERM_PROGRAM first and only fall back to TERM in case TERM_PROGRAM isn't set (e.g. over SSH).  And to avoid any misunderstandings I'd like to have this documented, i.e. that mdcat only looks at TERM for wezterm if TERM_PROGRAM didn't yield a good result.

[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply mdcat.1.adoc:105]: > I'd like to avoid the impression that you need to set both TERM and TERM_PROGRAM to identify wezterm

No idea what to write. Please just commit what you want to see. I don't have time to bikesheed over this documentation part. I might as well just cherry-pick my first commit in nixpkgs.

[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply mdcat.1.adoc:105]:
```suggestion
    If this variable is `wezterm`, mdcat assumes that the terminal is WezTerm. By default WezTerm does not set TERM to wezterm but mdcat can still detect it trough TERM_PROGRAM.
```

[the remark this answers is not in the document]
Comment (lunaryorn) [reply mdcat.1.adoc:105]: Nor is the opposite written anywhere, so I'd like to have the facts spelled out explicitly.

Whether we can make progress here depends on you entirely.  I hold not stake here; you're asking me to merge a change on your behalf, not on mine, so I do not consider it my responsibility to amend these changes.  I'm merely stating my terms, and it's my way or the high way.

Please excuse me for being blunt, but I think you remember what you wrote before deleting it, and while I do give you credit for deleting it, I have read it nonetheless (remember, Github sends mail notifications for _every_ comment), so you'll understand that I'm less positively inclined towards this conversation now.

Commit your suggestion and I'll merge this pull request, and then you can go back to cherry-picking your nixpkg commits.

**Review by lunaryorn — APPROVED** [review]

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (lunaryorn) [reply CHANGELOG.md]: Would you mind to commit this suggestion as well?

[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply CHANGELOG.md]: I applied it via the git client and not the web so it is shown as not applied.
